include/sound/sof/control.h

```diff
@@ -149,7 +149,9 @@ struct sof_ipc_comp_event {
 	/* control data - add new types if needed */
 	union {
 		/* data can be used by binary controls */
-		struct sof_abi_hdr data[0];
```

Comment:
@jsarha these were left alone to keep them consistent with the structure definitions in the FW.

Reply:
I do not think it makes much difference which definition is used. In any case a direct reference to an array element, beyond the first one, does not work. But I can make a corresponding change in sof side too, if this commit is otherwise approved.